### PR TITLE
mkinitcpio: Fix `lvm2` hook after `lvmetad` suppression from `lvm2`

### LINKED ISSUE #53496: `lvm2` hook build failure with `mkinitcpio`
### Is this a new report?

Yes

### System Info

Void 6.1.51-mod-std x86_64-musl GenuineIntel uptodate rF

### Package(s) Affected

mkinitcpio-39.2_2, lvm2-2.03.23_1

### Does a report exist for this bug with the project's home (upstream) and/or another distro?

_No response_

### Expected behaviour

Should build the `lvm2` initramfs hook without error

### Actual behaviour

```
Executing post-install kernel hook: 20-initramfs ...
==> Starting build: '6.6.64_1'
  -> Running build hook: [base]
  -> Running build hook: [udev]
  -> Running build hook: [autodetect]
  -> Running build hook: [keyboard]
==> WARNING: Possibly missing firmware for module: 'xhci_pci'
  -> Running build hook: [modconf]
  -> Running build hook: [keymap]
  -> Running build hook: [consolefont]
==> WARNING: consolefont: no font found in configuration
  -> Running build hook: [block]
  -> Running build hook: [mdadm_udev]
Custom /etc/mdadm.conf file will be used in initramfs for assembling arrays.
  -> Running build hook: [netconf]
  -> Running build hook: [dropbear]
dropbear_rsa_host_key : SHA256:xxwFhN0i6LxdVPWbuCkLqXlhodqRpRrPPB+k08g2IRo
dropbear_ecdsa_host_key : SHA256:IohkvDwO/ga8uLUWXfzfwCmtfZlVF+GEcOOXwFclUiE
dropbear_ed25519_host_key : SHA256:+GVTznsuYKT6RfZc7MJcBTPayJg367AqCbp+lC762Dg
  -> Running build hook: [encryptssh]
==> WARNING: Possibly missing firmware for module: 'crypto_safexcel'
==> WARNING: Possibly missing firmware for module: 'qat_c3xxx'
==> WARNING: Possibly missing firmware for module: 'qat_c62x'
  -> Running build hook: [lvm2]
==> ERROR: binary not found: 'lvmetad'
  -> Running build hook: [filesystems]
  -> Running build hook: [fsck]
==> Generating module dependencies
==> Creating zstd-compressed initcpio image: '/boot/initramfs-6.6.64_1.img'
==> WARNING: errors were encountered during the build. The image may not be complete.
```

### Steps to reproduce

1. Install and use `mkinitcpio` as initramfs :
```sh
sudo xbps-install mkinitcpio mkinitcpio-lvm2 lvm2
xbps-alternatives -s mkinitcpio
```
2. Regenerate iniramfs `sudo xbps-reconfigure -fa `
3. See that hook `lvm2` is built without error

### PR DESCRIPTION
**Pull request by MonsieurV**
#### Testing the changes
- I tested the changes in this PR: **briefly**

The changes where tested on server, I was able to make the hook `lvm2` build again with `sudo xbps-reconfigure -fa `.
It was indeed allowing to detect LVM partitions in the initramfs.
However I didn't succeed to make a complete boot (from initramfs to Void) for my configuration, but I think this to other issues (I was trying to setup up a server with disk encryption, there where multiple moving part and I was shooting in the dark debugging with a poor KVM). I don't have time for now to test it on my laptop (where I run Void with LVM on LUKS). Ideally someone using LVM would make a test.

In all cases, it works better as is than when not building at all the `lvm2` hook.

#### Local build testing
- I built this PR locally for my native architecture, (`x86_64-musl`)

PS. I think https://github.com/void-linux/void-packages/tree/master/srcpkgs/lvm2/files/lvmetad can be deleted from `lvm2`, as the binary doesn't exist anymore.

Closes #53496

### CODE REVIEW

File: srcpkgs/mkinitcpio/files/69-dm-lvm.rules [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Copyright (C) 2012,2021 Red Hat, Inc. All rights reserved.
```
Comment (MonsieurV) [inline srcpkgs/mkinitcpio/files/69-dm-lvm.rules:1]: I copied the file from Arch Linux udev. It seems near enough from `69-dm-lvm-metad.rules`, without the `LVMetaD` bits

File: srcpkgs/mkinitcpio/files/lvm2_hook [hunk 2]
```diff
@@ -1,24 +0,0 @@
-#!/usr/bin/ash
-
-run_earlyhook() {
-    mkdir /run/lvm
```
Comment (MonsieurV) [inline srcpkgs/mkinitcpio/files/lvm2_hook:4]: I was not totally sure was is the purpose of this `mkir`. If it is not related to running `lvmetad`, it should be restored. The other parts are for `lvmetad` and can be dropped, else the hook will fail.

**Review by MonsieurV — COMMENTED** [review]
Some motivation for the changes

File: srcpkgs/mkinitcpio/files/69-dm-lvm.rules (hunk 1 above)
Comment (classabbyamp) [reply srcpkgs/mkinitcpio/files/69-dm-lvm.rules:1]: a file with the same name is shipped with the `lvm2` package already. is it significantly different? mkinitcpio hooks can install udev rules from the system too

File: srcpkgs/mkinitcpio/template [hunk 3]
```diff
@@ -52,8 +53,6 @@ mkinitcpio-lvm2_package() {
 	depends="${sourcepkg}>=${version}_${revision} lvm2 thin-provisioning-tools"
 	short_desc+=" - lvm2 support"
 	pkg_install() {
```
Comment (classabbyamp) [inline srcpkgs/mkinitcpio/template:55]:
```suggestion
	pkg_install() {
		vmove usr/lib/initcpio/udev/69-dm-lvm.rules
```

File: srcpkgs/mkinitcpio/files/69-dm-lvm.rules (hunk 1 above)
Comment (classabbyamp) [reply srcpkgs/mkinitcpio/files/69-dm-lvm.rules:1]: a file with the same name is installed by `lvm2`. is it significantly different?

**Comment by classabbyamp** [comment]
there's probably some useful info in this thread: https://gitlab.archlinux.org/archlinux/mkinitcpio/mkinitcpio/-/issues/275

**Comment by MonsieurV** [comment]
I've tested it on my laptop (LVM on LUKS), switching from Dacrut to mkinitcpio, I can confirm it works.

**Comment by classabbyamp** [comment]
ok, I talked with upstream about these udev rules. They really should be vendored in the mkinitcpio tree, so I've started that process: https://gitlab.archlinux.org/archlinux/mkinitcpio/mkinitcpio/-/merge_requests/416

I adapted what you did, but also discovered that we don't need the lvm2 patch at all: #53573 

I was able to successfully use the lvm2 hook to boot a VM (without encryption). Thanks for looking into this, I'll get it all merged soon.

**Pull request closed without merging**